lib/kubeclient/common.rb

```diff
@@ -39,6 +39,12 @@ module ClientMixin
       'fieldSelector' => :field_selector
     }.freeze
 
+    DELETE_ARGUMENTS = {
+      'gracePeriodSeconds' => :grace_period_seconds,
+      'orphanDependents'   => :orphan_dependents,
```

Comment:
this is a deprecated param starting 1.7. propagation policy should be used. moreover, it cannot be set together with propagationPolicy so shall it be supported here, there should be a validation they are not used together. (or at least verification that the server is validating this and returning a clear error)